lightning/src/routing/router.rs

```diff
@@ -63,6 +63,15 @@ impl Readable for InFlightHtlcs {
 	}
 }
 
+/// A trait defining behavior for routing an payment.
+pub trait Router {
```

Comment:
I know many of the types in here are ordered inside-out instead of outside-in, but could you move this immediately above `InFlightHtlcs` instead? Someone reading the file probably would want to see `Router` before `InFlightHtlcs`.